source/common/api/os_sys_calls_impl_linux.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "envoy/api/os_sys_calls_linux.h"
+
+#include "common/api/os_sys_calls_impl.h"
+#include "common/singleton/threadsafe_singleton.h"
+
+namespace Envoy {
+namespace Api {
+
+class LinuxOsSysCallsImpl : public OsSysCalls, LinuxOsSysCalls {
```

Comment:
Is there value in embedding and calling through? It seems like we could just have 2 syscall impls that are used?

Reply:
Ok, I changed this to work the way you suggested. I was originally thinking that maybe it would make sense that there would be only one `*OsSysCalls` version for each platform, which would support all the system calls available for the platform. I'm totally fine with either way though.